File: random_code/A. - Геймер Хемосе.cpp
#include <iostream>
#include <vector>
#include <algorithm>

int main() {
    int tests;
    std::cin >> tests;

    for (int i = 0; i < tests; ++i) {
        int n, H;
        std::cin >> n >> H;

        std::vector<int> damage(n);
        for (int j = 0; j < n; ++j) {
            std::cin >> damage[j];
        }

        std::sort(damage.begin(), damage.end());
        int x = damage[n - 1];
        int y = damage[n - 2];
        int t = H % (x + y);
        int m = H / (x + y);
        int ans;

        if (t == 0) {
            ans = 2 * m;
        } else if (t <= x) {
            ans = 2 * m + 1;
        } else {
            ans = 2 * m + 2;
        }

        std::cout << ans << std::endl;
    }

    return 0;
}

File: random_code/A. Геймер Хемосе.py
def main():
    n, h = map(int, input().split())
    damage = list(map(int, input().split()))
    damage.sort()
    x = damage[-1]
    y = damage[-2]
    t = h % (x + y)
    m = h // (x + y)
    if t == 0:
        ans = 2 * m
    elif t <= x:
        ans = 2 * m + 1
    else:
        ans = 2 * m + 2
    return ans


if __name__ == "__main__":
    for _ in range(int(input())):
        print(main())
